[29.x] backports and rc2

glozow opened the pull request:
backports:
- #32049
- #32063
- #32064 
- #32070 
- #31917
- https://github.com/bitcoin-core/gui/pull/858

l0rinc committed:
fuzz: Always restrict base conversion input lengths

They seem to cause timeouts:
> Issue 397734700: bitcoin-core:base58check_encode_decode: Timeout in base58check_encode_decode

The `encoded_string.empty()` check was corrected here to `decoded.empty()` to make sure the `(0, decoded.size() - 1)` range is always valid.

Github-Pull: #31917
Rebased-From: bad1433ef2b5b02ac4b1c6c1d9482c513e5b2192

Co-authored-by: maflcko <6399679+maflcko@users.noreply.github.com>
Co-authored-by: marcofleon <marleo23@proton.me>
Co-authored-by: Martin Zumsande <mzumsande@gmail.com>

l0rinc committed:
fuzz: make sure DecodeBase58(Check) is called with valid values more often

In Base58 fuzz the two roundtrips are merged now, the new `decode_input` switches between a completely random input and a valid encoded one, to make sure the decoding passes more often.
The `max_ret_len` can also exceed the original length now and is being validated more thoroughly.

Github-Pull: #31917
Rebased-From: d5537c18a9034647ba4c9ed4008abd7fee33989e

Co-authored-by: maflcko <6399679+maflcko@users.noreply.github.com>
Co-authored-by: marcofleon <marleo23@proton.me>

davidgumberg committed:
contrib: Fix `gen-bitcoin-conf.sh`.

In #31118, the format of bitcoind's `--help` output changed slightly in
a way that breaks `gen-bitcoin-conf.sh`, modify the script to accomodate
the new format, by starting after the line that says "Options:" and
strip the `-help` option and its description from the output.

Github-Pull: #32049
Rebased-From: a24419f8bed5e1145ce171dbbdad957750585471

glozow committed:
[doc] update example bitcoin.conf with missing options

davidgumberg commented:
lgtm ACK https://github.com/bitcoin/bitcoin/pull/32062/commits/188ec87589476355a4280f8a561d954b9fbbb7af

mzumsande committed:
test: fix intermittent failure in p2p_orphan_handling.py

If we bump the mocktime before the node has successfully disconnected
the peer, the requests for both parents could be spread over
two GETDATAS, which would make the test fail.

Github-Pull: #32063
Rebased-From: 02942056fd861581503a8a35a06dcf22d4ba1473

achow101 committed:
build: Remove manpages when making MacOS app

Github-Pull: #32064
Rebased-From: 80b5e7f2cb7fbfbd724e1f52b00c0e72b79a200b

Sjors committed:
build: use make < 3.82 syntax for define directive

From the GNU make 3.82 release announcement:

* The 'define' make directive now allows a variable assignment operator
  after the variable name, to allow for simple, conditional, or appending
  multi-line variable assignment.

macOS ships with 3.81. This caused the multiprocess config options
to be ignored.

Fixes #32068

Github-Pull: #32070
Rebased-From: 9157d9e449870851ef455e077249ac46fc2df24c

Co-authored-by: Ryan Ofsky <ryan@ofsky.org>

theStack committed:
qt: doc: adapt outdated binary paths to CMake changes

Github-Pull: bitcoin-core/gui#858
Rebased-From: 7ebc458a8cb994bc3c0c129da61353968d955bc2

glozow committed:
[build] bump to 29.0rc2

glozow committed:
[doc] update man pages for 29.0rc2

fanquake commented:
Can also pull in https://github.com/bitcoin-core/gui/pull/858 (if not this rc than next).

Sjors commented:
ACK f86997eb03d825b7c963adb1463ba83105628a74

I cherry-picked the pull requests, ran `gen-bitcoin-conf.sh` (built with ` -DWITH_ZMQ=ON`) and compared to this PR (minus the version bump commits).

I do get a slightly difference in `share/examples/bitcoin.conf`, maybe due to a build option?

```
-# estimatefee, http, i2p, ipc, leveldb, libevent, mempool,
+# estimatefee, http, i2p, ipc, leveldb, libevent, lock, mempool,
```

glozow commented:
> Can also pull in [bitcoin-core/gui#858](https://github.com/bitcoin-core/gui/pull/858) (if not this rc than next).

added

Sjors commented:
ACK 74df31cb0bdef9cce31ae62ed71a1e386cba0274

Don't forget to add https://github.com/bitcoin-core/gui/pull/858 to the PR description.

hebasto reviewed (approved):
ACK 74df31cb0bdef9cce31ae62ed71a1e386cba0274, I have reviewed the code and it looks OK.

I've backported PRs and generated `contrib/devtools/gen-bitcoin-conf.sh` locally and got zero-diff with this PR branch.

ismaelsadeeq reviewed (commented):
Code review ACK 74df31cb0bdef9cce31ae62ed71a1e386cba0274

I've verified that the patch-ids are identical. Only 4e438d326ea55ac0f98f89e41e69b56354e801e7 has a different patch-id from the original commit, and that's due to line number differences. 

I've verified that the backported commits is identical to the PR commits.
